### ISSUE TITLE
How can I use nested cross validation with hyperopt?

### ISSUE THREAD

**Issue by Marmeladenbrot** [hyperopt/hyperopt#707]
I use LightGBMRegressor via sklearn API.

I want to choose the best hyperparameters via hyperopt while getting a better generalization error than I get via k-fold cross validation.

Every example I've seen uses GridSearchCV ....

How can I use hyperopt + nested CV?

**Comment by Arktius** [comment]
That's a nice tutorial: https://github.com/hyperopt/hyperopt/wiki/FMin

You have to put all your code inside a function. Your code would look somehow like this:

def f(parameters)

- allmetrics = []

- Split Data
- Iterate over all k folds
- LightGBMRegressor via sklearn API
- Calculate your loss / accuracy and add it to allmetrics.append([loss,acc])

Now you've logged the loss and the acc for each iteration. Take the mean of all loss and all accuracies so that you end up with 2 numbers. 

- Return Loss


fmin( f,....)

**Comment by Arktius** [comment]
https://www.kaggle.com/ilialar/hyperparameters-tunning-with-hyperopt#Hyperopt
